src/main/java/address/browser/AddressBookBrowser.java

```diff
@@ -0,0 +1,273 @@
+        loadProfilePages(listOfPersonsToBeLoaded);
+        selectTabAssignedForPerson(person);
+    }
+
+    /**
+     * Computes the range of index of the displayed AddressBook contact that needs to be added to the browser cache.
+     * @param listOfContactsDisplayed The list of contacts currently displayed.
+     * @param indexOfPersonInListOfContacts The index of the currently selected person in the AddressBook contact list.
+     * @return
+     */
+    private Range getRangeOfIndexToLoad(List<Person> listOfContactsDisplayed, int indexOfPersonInListOfContacts) {
+        Range rangeOfIndexToLoad;
+        if(BrowserManager.NUMBER_OF_PRELOADED_PAGE >= listOfContactsDisplayed.size()){
+            rangeOfIndexToLoad = new Range(indexOfPersonInListOfContacts, indexOfPersonInListOfContacts -1);
+        } else {
+            rangeOfIndexToLoad = new Range(indexOfPersonInListOfContacts,
+                    (indexOfPersonInListOfContacts + BrowserManager.NUMBER_OF_PRELOADED_PAGE - 1)
+                            % listOfContactsDisplayed.size());
+        }
+        return rangeOfIndexToLoad;
+    }
+
+    /**
+     * Gets a list of Person objects that are already loaded in the browser cache.
+     * @param listOfContactsDisplayed A list of contacts displayed on the UI.
+     * @param rangeOfIndexToLoad A range of index of the listOfContactsDisplayed that are to be loaded into the browser
+     *                           cache.
+     * @return a list of Person objects that are already loaded in the browser cache.
+     */
+    private ArrayList<Person> getListOfAlreadyLoadedPerson(List<Person> listOfContactsDisplayed,
+                                                           Range rangeOfIndexToLoad) {
+        ArrayList<Person> listOfAlreadyLoadedPerson = new ArrayList<>();
+        pages.stream().forEach(browserTab -> {
+                if (rangeOfIndexToLoad.isWithinRange(listOfContactsDisplayed.indexOf(browserTab.getPerson()))){
+                    listOfAlreadyLoadedPerson.add(browserTab.getPerson());
+                }
+            });
+        return listOfAlreadyLoadedPerson;
+    }
+
+    /**
+     * Unload profile pages that are unneeded from the cache of the browser.
+     * @param listOfContactsDisplayed A list of contacts displayed on the UI.
+     * @param rangeOfIndexToLoad A range of index of the listOfContactsDisplayed that are to be loaded into the browser
+     *                           cache.
+     */
+    private void unloadProfilePages(List<Person> listOfContactsDisplayed, Range rangeOfIndexToLoad) {
+
+        for (BrowserTab browserTab: pages) {
+            int personIndexInFilteredPerson = listOfContactsDisplayed.indexOf(browserTab.getPerson());
+
+            if (personIndexInFilteredPerson == -1) {
+                continue;
+            }
+
+            if (!rangeOfIndexToLoad.isWithinRange(personIndexInFilteredPerson)) {
+                browserTab.unloadProfilePage();
+            }
```

Comment:
Instead of using range which is hard to understand to determine who we would like to load, why don't this method just pass the list of `Person` that we would like to load to this method, and then unload any tabs which `Person` is not inside the list?

We can use `stream.map` for that, too.